### ISSUE TITLE
Learn button throws unhandled error

### ISSUE THREAD

**Issue by krharrison1** [kiwi-cam/homebridge-broadlink-rm#320]
**Describe the bug**
The LEARN button on the HB accessories page does not work correctly; it throws an error when pressed. I discovered this while running 4.4.3 of the plug-in; upgraded to v4.4.4-beta.3 and the issue persists. 

From HB logs:
[5/18/2021, 4:12:38 PM] [homebridge-broadlink-rm-pro] This plugin threw an error from the characteristic 'On': Unhandled error thrown inside write handler for characteristic: log is not a function. See https://git.io/JtMGR for more info.

**To Reproduce**
Press the LEARN button and look at the logs. 

**Expected behavior**
Press the LEARN button in HB and the RF/IR code learning process begins.

**Screenshots**
[5/18/2021, 4:12:38 PM] [homebridge-broadlink-rm-pro] This plugin threw an error from the characteristic 'On': Unhandled error thrown inside write handler for characteristic: log is not a function. See https://git.io/JtMGR for more info.

**Desktop (please complete the following information):**
Windows, Chrome

**Comment by kiwi-cam** [comment]
I've tried everything I can think of and I can't recreate this issue.  I wonder if it's a dependant module that is creating the error, could you try running `npm list -g --depth=1` and share the output.

It'll look something like this:
```
/usr/lib
...
├─┬ homebridge-broadlink-rm-pro@4.4.4-beta.3
│ ├── chai@4.3.4
│ ├── compare-versions@3.6.0
│ ├── fakegato-history@0.6.1
│ ├── find-key@2.1.3
│ ├── github-version-checker@2.2.0
│ ├── kiwicam-broadlinkjs-rm@0.9.14-beta.2
│ ├── mqtt@4.2.6
│ ├── node-arp@1.0.6
│ ├── node-persist@2.1.0
│ ├── ping@0.3.0
│ ├── semver@7.3.5
│ └── uuid@8.3.2
...
```

**Comment by krharrison1** [comment]
Thanks for your quick response. I think it has to be some dependent module because I'd expect more people would be seeing this, if otherwise.Here's the output:

/home/pi/.nvm/versions/node/v13.11.0/lib
├─┬ npm@6.14.4
│ ├── abbrev@1.1.1
│ ├── ansicolors@0.3.2
│ ├── ansistyles@0.1.3
│ ├── aproba@2.0.0
│ ├── archy@1.0.0
│ ├── bin-links@1.1.7
│ ├── bluebird@3.5.5
│ ├── byte-size@5.0.1
│ ├── cacache@12.0.3
│ ├── call-limit@1.1.1
│ ├── chownr@1.1.4
│ ├── ci-info@2.0.0
│ ├── cli-columns@3.1.2
│ ├── cli-table3@0.5.1
│ ├── cmd-shim@3.0.3
│ ├── columnify@1.5.4
│ ├── config-chain@1.1.12
│ ├── debuglog@1.0.1
│ ├── detect-indent@5.0.0
│ ├── detect-newline@2.1.0
│ ├── dezalgo@1.0.3
│ ├── editor@1.0.0
│ ├── figgy-pudding@3.5.1
│ ├── find-npm-prefix@1.0.2
│ ├── fs-vacuum@1.2.10
│ ├── fs-write-stream-atomic@1.0.10
│ ├── gentle-fs@2.3.0
│ ├── glob@7.1.6
│ ├── graceful-fs@4.2.3
│ ├── has-unicode@2.0.1
│ ├── hosted-git-info@2.8.8
│ ├── iferr@1.0.2
│ ├── imurmurhash@0.1.4
│ ├── infer-owner@1.0.4
│ ├── inflight@1.0.6
│ ├── inherits@2.0.4
│ ├── ini@1.3.5
│ ├── init-package-json@1.10.3
│ ├── is-cidr@3.0.0
│ ├── json-parse-better-errors@1.0.2
│ ├── JSONStream@1.3.5
│ ├── lazy-property@1.0.0
│ ├── libcipm@4.0.7
│ ├── libnpm@3.0.1
│ ├── libnpmaccess@3.0.2
│ ├── libnpmhook@5.0.3
│ ├── libnpmorg@1.0.1
│ ├── libnpmsearch@2.0.2
│ ├── libnpmteam@1.0.2
│ ├── libnpx@10.2.2
│ ├── lock-verify@2.1.0
│ ├── lockfile@1.0.4
│ ├── lodash._baseindexof@3.1.0
│ ├── lodash._baseuniq@4.6.0
│ ├── lodash._bindcallback@3.0.1
│ ├── lodash._cacheindexof@3.0.2
│ ├── lodash._createcache@3.1.2
│ ├── lodash._getnative@3.9.1
│ ├── lodash.clonedeep@4.5.0
│ ├── lodash.restparam@3.6.1
│ ├── lodash.union@4.6.0
│ ├── lodash.uniq@4.5.0
│ ├── lodash.without@4.4.0
│ ├── lru-cache@5.1.1
│ ├── meant@1.0.1
│ ├── mississippi@3.0.0
│ ├── mkdirp@0.5.4
│ ├── move-concurrently@1.0.1
│ ├── node-gyp@5.1.0
│ ├── nopt@4.0.1
│ ├── normalize-package-data@2.5.0
│ ├── npm-audit-report@1.3.2
│ ├── npm-cache-filename@1.0.2
│ ├── npm-install-checks@3.0.2
│ ├── npm-lifecycle@3.1.4
│ ├── npm-package-arg@6.1.1
│ ├── npm-packlist@1.4.8
│ ├── npm-pick-manifest@3.0.2
│ ├── npm-profile@4.0.4
│ ├── npm-registry-fetch@4.0.3
│ ├── npm-user-validate@1.0.0
│ ├── npmlog@4.1.2
│ ├── once@1.4.0
│ ├── opener@1.5.1
│ ├── osenv@0.1.5
│ ├── pacote@9.5.12
│ ├── path-is-inside@1.0.2
│ ├── promise-inflight@1.0.1
│ ├── qrcode-terminal@0.12.0
│ ├── query-string@6.8.2
│ ├── qw@1.0.1
│ ├── read@1.0.7
│ ├── read-cmd-shim@1.0.5
│ ├── read-installed@4.0.3
│ ├── read-package-json@2.1.1
│ ├── read-package-tree@5.3.1
│ ├── readable-stream@3.6.0
│ ├── readdir-scoped-modules@1.1.0
│ ├── request@2.88.0
│ ├── retry@0.12.0
│ ├── rimraf@2.7.1
│ ├── safe-buffer@5.1.2
│ ├── semver@5.7.1
│ ├── sha@3.0.0
│ ├── slide@1.1.6
│ ├── sorted-object@2.0.1
│ ├── sorted-union-stream@2.1.3
│ ├── ssri@6.0.1
│ ├── stringify-package@1.0.1
│ ├── tar@4.4.13
│ ├── text-table@0.2.0
│ ├── tiny-relative-date@1.3.0
│ ├── uid-number@0.0.6
│ ├── umask@1.1.0
│ ├── unique-filename@1.1.1
│ ├── unpipe@1.0.0
│ ├── update-notifier@2.5.0
│ ├── uuid@3.3.3
│ ├── validate-npm-package-license@3.0.4
│ ├── validate-npm-package-name@3.0.0
│ ├── which@1.3.1
│ ├── worker-farm@1.7.0
│ └── write-file-atomic@2.4.3
├─┬ npm-check-updates@4.0.6
│ ├── chalk@3.0.0
│ ├── cint@8.2.1
│ ├── cli-table@0.3.1
│ ├── commander@5.0.0
│ ├── fast-diff@1.2.0
│ ├── find-up@4.1.0
│ ├── get-stdin@7.0.0
│ ├── json-parse-helpfulerror@1.0.3
│ ├── libnpmconfig@1.2.1
│ ├── lodash@4.17.15
│ ├── node-alias@1.0.4
│ ├── pacote@11.1.4
│ ├── progress@2.0.3
│ ├── prompts@2.3.2
│ ├── rc-config-loader@3.0.0
│ ├── requireg@0.2.2
│ ├── semver@7.1.3
│ ├── semver-utils@1.1.4
│ ├── spawn-please@0.3.0
│ └── update-notifier@4.1.0
└─┬ pm2@4.2.3
  ├── @pm2/agent@0.5.26
  ├── @pm2/io@4.3.3
  ├── @pm2/js-api@0.5.60
  ├── @pm2/pm2-version-check@1.0.3
  ├── async@3.2.0
  ├── blessed@0.1.81
  ├── chalk@2.4.2
  ├── chokidar@3.3.1
  ├── cli-table-redemption@1.0.1
  ├── commander@2.15.1
  ├── cron@1.7.1
  ├── date-fns@1.30.1
  ├── debug@4.1.1
  ├── enquirer@2.3.4
  ├── eventemitter2@5.0.1
  ├── fclone@1.0.11
  ├── lodash@4.17.14
  ├── mkdirp@0.5.1
  ├── moment@2.24.0
  ├── needle@2.4.0
  ├── pidusage@2.0.17
  ├── pm2-axon@3.3.0
  ├── pm2-axon-rpc@0.5.1
  ├── pm2-deploy@0.4.0
  ├── pm2-multimeter@0.1.2
  ├── promptly@2.2.0
  ├── ps-list@6.3.0
  ├── semver@5.7.1
  ├── shelljs@0.8.3
  ├── source-map-support@0.5.12
  ├── sprintf-js@1.1.2
  ├── systeminformation@4.23.1
  ├── vizion@2.0.2
  └── yamljs@0.3.0

**Comment by krharrison1** [comment]
Updates I've made:
- sudo apt update , sudo apt full-upgrade to update all packages on my pi
- HB and all plugins are on latest version
- sudo hb-service update-node

The error persists, learn button still throws an error:
[5/23/2021, 8:40:17 AM] [homebridge-broadlink-rm-pro] This plugin threw an error from the characteristic 'On': Unhandled error thrown inside write handler for characteristic: log is not a function. See https://git.io/JtMGR for more info.

I'm stuck at this point - any ideas? I cannot find another way to capture the IR code that I need for a device command.

**Comment by aferrar** [comment]
Hi, I encountered this issue also but only after having it work for a period of time prior. I stepped through each parameter I had changed after it stopped working and was able to isolate the issue to the "type" parameter of the test accessory. 

The error gets thrown whenever the "type" was set to "tv" and the error went away when the "type" was changed from "tv" to "switch". @krharrison1 , give that a try and see if it fixes things for you.

@kiwi-cam , I hope this is helpful and reproducible. Also, thanks a ton for maintaining this project!
